src/actions/auth.js

```diff
@@ -0,0 +1,75 @@
+import jwtDecode from 'jwt-decode';
+import {SubmissionError} from 'redux-form';
+
+import {API_BASE_URL} from '../config';
+import {handleResponse} from './utils';
+import {saveAuthToken, clearAuthToken} from '../local-storage';
+
+export const SET_AUTH_TOKEN = 'SET_AUTH_TOKEN';
+export const setAuthToken = authToken => ({
+    type: SET_AUTH_TOKEN,
+    authToken
+});
+
+export const SET_CURRENT_USER = 'SET_CURRENT_USER';
+export const setCurrentUser = currentUser => ({
+    type: SET_CURRENT_USER,
+    currentUser
+});
+
+
+// Stores the auth token in state and localStorage, and decodes and stores
+// the user data stored in the token
+const storeAuthInfo = (authToken, dispatch) => {
+    const decodedToken = jwtDecode(authToken);
+    dispatch(setAuthToken(authToken));
+    dispatch(setCurrentUser(decodedToken.user));
+    saveAuthToken(authToken);
+};
+
+export const login = (username, password) => dispatch => {
+    const token = btoa(`${username}:${password}`);
```

Comment:
`btoa` won't be familiar. probably should add a comment in code, and also mention in walk through.